common/lc_library.cpp

```diff
@@ -1670,6 +1708,464 @@ bool lcPiecesLibrary::LoadPrimitive(lcLibraryPrimitive* Primitive)
 	return true;
 }
 
+void lcPiecesLibrary::SetLegoStyleStudFiles(bool Enable, bool Reload)
+{
+	auto SetFile = [this](QStringList &Files, const QStringList &Content, const QString &FileName)
+	{
+		const QLatin1String LineEnding("\r\n");
+		QByteArray FileData;
+
+		foreach (const QString &String, Content)
+		{
+			FileData.append(String.toLatin1().constData() + LineEnding);
+		}
+
+		mLegoStyleDisplayStudFiles.emplace(std::make_pair(FileName.toStdString(), FileData));
+
+		Files.append(FileName);
+	};
+
+	QString ColorCode = Enable ? "4242" : "16";
+
+	QStringList Content, Files;;
+	Content << QString("0 Stud Tube Open 0.5");
```

Comment:
I think these parts could be loaded from Qt resources. Let me look into doing something like that for the stud logos.

Reply:
Certainly, this is my thinking also. However, to treat switching on and off this option without reloading the library, I’ve added the stud colour as a variable. As a Qt resource it will be necessary to find a solution to switch this option on and off - perhaps two sets of primitives?

Reply:
I've checked in the stud logo primitives in a zip file, but now I see that those were easier because they are new primitives and don't replace anything. I need a better system so we can add/remove zip files.

Maybe it will also take care of all the special cases we have for both the stud logos and lego style studs.

Reply:
Look at a8d757c0c4344dcbbcc8fb0541c8fe50dc2458ef, it adds a way to override stud.dat by loading a higher priority library zip file with only the stud primitives. I think something similar can be done for your change.

Reply:
Excellent. I'll take a look.

Cheers,

Reply:
The new sources behaviour is very nice and will make it much easier to add missing 'LEGO-Style' primitives.

Cheers,